### ISSUE TITLE
v4 - Some grid docs examples still use defunct .visible-xs-block

### ISSUE THREAD

**Issue by monoblaine** [twbs/bootstrap#17263]
According to the [migration guide](http://v4-alpha.getbootstrap.com/migration/#responsive-utilities), the `visible-*-block` classes are gone:

> The old classes (`.hidden-xs` `.hidden-sm` `.hidden-md` `.hidden-lg` `.visible-xs-block` `.visible-xs-inline` `.visible-xs-inline-block` `.visible-sm-block` `.visible-sm-inline` `.visible-sm-inline-block` `.visible-md-block` `.visible-md-inline` `.visible-md-inline-block` `.visible-lg-block` `.visible-lg-inline` `.visible-lg-inline-block`) are gone.

However, [the grid example here](http://v4-alpha.getbootstrap.com/layout/grid/#example-mobile-tablet-desktop) still uses this technique:

```html
<div class="clearfix visible-xs-block"></div>
```

Am I missing something?

**Comment by cvrebert** [comment]
No, it probably just escaped getting updated.

**Comment by monoblaine** [comment]
So, in order to achieve the same goal using the new responsive utilities, we need to write something like this:

```html
<div class="row">
    <div class="col-sm-6 col-md-4 col-lg-3">.col-sm-6 .col-md-4 .col-lg-3</div>
    <div class="col-sm-6 col-md-4 col-lg-3">.col-sm-6 .col-md-4 .col-lg-3</div>
    <div class="clearfix hidden-xs-down hidden-md-up"></div> <!-- Show only for sm -->
    <div class="col-sm-6 col-md-4 col-lg-3">.col-sm-6 .col-md-4 .col-lg-3</div>
    <div class="clearfix hidden-sm-down hidden-lg-up"></div> <!-- Show only for md -->
    <div class="col-sm-6 col-md-4 col-lg-3">.col-sm-6 .col-md-4 .col-lg-3</div>
    <div class="clearfix hidden-md-down hidden-xl-up"></div> <!-- Show only for lg -->
    <div class="col-sm-6 col-md-4 col-lg-3">.col-sm-6 .col-md-4 .col-lg-3</div>
</div>
```

Don't you think the following code is much easier to write and understand?

```html
<div class="row">
    <div class="col-sm-6 col-md-4 col-lg-3">.col-sm-6 .col-md-4 .col-lg-3</div>
    <div class="col-sm-6 col-md-4 col-lg-3">.col-sm-6 .col-md-4 .col-lg-3</div>
    <div class="clearfix visible-sm-block"></div> <!-- Show only for sm -->
    <div class="col-sm-6 col-md-4 col-lg-3">.col-sm-6 .col-md-4 .col-lg-3</div>
    <div class="clearfix visible-md-block"></div> <!-- Show only for md -->
    <div class="col-sm-6 col-md-4 col-lg-3">.col-sm-6 .col-md-4 .col-lg-3</div>
    <div class="clearfix visible-lg-block"></div> <!-- Show only for lg -->
    <div class="col-sm-6 col-md-4 col-lg-3">.col-sm-6 .col-md-4 .col-lg-3</div>
</div>
```

**Comment by cvrebert** [comment]
In that particular case, yes. Every design has some trade-off.
Clearfixes aside, the "only show on exactly one screen size" case is believed to be relatively less common than the "screen size X and bigger/smaller" cases, which previously required using multiple classes. The new approach is also more robust and easier to use in the very common case where some other class sets the `display`, e.g.
```css
.foo {
  display: inline-block;
  /*...*/
}
```
```html
<div class="foo hidden-sm-down">...</div><!--Not enough space for this on small screens-->
```
Previously you'd have to "just know" what `display` value `.foo` sets and use the corresponding visibility class, whereas the new classes don't require that and are generic with respect to the non-`none` `display` value.

**Comment by monoblaine** [comment]
Just for the record, I love the new responsive utilities. I was already using my own implementation in v3: #17124 :) And you're definitely right about that specific clearfix case being relatively less common. Keeping the old `.visible-*-block` classes just for this purpose would be weird and confusing. How about some clearfix-specific responsive utility classes? (`.clearfix-sm`, `.clearfix-md`, etc.)

**Comment by cvrebert** [comment]
> How about some clearfix-specific responsive utility classes? (`.clearfix-sm`, `.clearfix-md`, etc.)

I don't personally feel strongly about that one way or the other. I'd recommend you file a new issue for that feature request so that mdo can evaluate it.

**Comment by cvrebert** [comment]
Fixed by #17293. Thanks for reporting this!
